### PR TITLE
Form hook tweaks

### PR DESCRIPTION
**Pull request by TheMallen**
This PR carries our a few tweaks to the hooks from `react-form` I noticed needed doing while building out some forms in web using the package. It also switches `react-form-state` and `react-form` to use `@shopify/predicates` instead of duplicating predicate logic.

The details of the changes to `react-form`'s hook behaviour are in `CHANGELOG.md`

### CODE REVIEW

File: packages/react-form/src/hooks/submit.ts [hunk 1]
```diff
@@ -16,24 +16,21 @@ export function useSubmit<T extends FieldBag>(
   const [submitting, setSubmitting] = useState(false);
   const [remoteErrors, setRemoteErrors] = useState([] as FormError[]);
 
-  const submit = useCallback(
-    async (event?: React.FormEvent) => {
-      if (event && event.preventDefault && !event.defaultPrevented) {
-        event.preventDefault();
-      }
+  async function submit(event?: React.FormEvent) {
```
Comment (TheMallen) [inline packages/react-form/src/hooks/submit.ts]: Rerendering a `Form` element is not expensive, so this being reference equal between renders is not a big deal. In addition, checking `fieldBag` meant that in most cases this would be regenerated each render anyway.

File: packages/predicates/CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## [1.1.0]
+
+### Added
+
+- `notNumericString`
```
Comment (lemonmade) [inline packages/predicates/CHANGELOG.md]: This is not a good changelog

File: packages/react-form/src/hooks/field/field.ts [hunk 3]
```diff
@@ -188,9 +194,17 @@ export function useField<Value = string>(
...
-    return {value: input, validates: () => undefined};
+  if (isFieldConfig(input)) {
+    return input;
   }
 
-  return input as FieldConfig<Value>;
+  return {value: input, validates: () => undefined};
+}
+
+function isFieldConfig<Value>(input: any): input is FieldConfig<Value> {
```
Comment (lemonmade) [inline packages/react-form/src/hooks/field/field.ts]: `unknown` is a better type in these cases

File: packages/react-form/src/hooks/field/field.ts [hunk 4]
```diff
@@ -188,9 +194,17 @@ export function useField<Value = string>(
...
+  return (
+    typeof input === 'object' &&
```
Comment (lemonmade) [inline packages/react-form/src/hooks/field/field.ts]: `input` can't be null, right?

File: packages/react-form/src/hooks/field/test/field.test.tsx [hunk 5]
```diff
@@ -14,7 +14,7 @@ describe('useField', () => {
           <input
             id="test-field"
             name="test-field"
-            value={field.value}
+            value={field.value as any}
```
Comment (lemonmade) [inline packages/react-form/src/hooks/field/test/field.test.tsx]: Bummer, why was this needed?

File: packages/react-form/src/hooks/submit.ts (hunk 1 above)
Comment (lemonmade) [reply packages/react-form/src/hooks/submit.ts]: I am not sure that it is safe to say that the thing this callback gets passed to is cheap to re-render, in the end you really have no idea what the cost is. Ideally we would memoize because the consumer can't, but if that's impossible, then that's fine.

File: packages/react-form/src/hooks/field/field.ts (hunk 4 above)
Comment (TheMallen) [reply packages/react-form/src/hooks/field/field.ts]: nope, at least not if they are using TS :P

File: packages/react-form/src/hooks/field/test/field.test.tsx (hunk 5 above)
Comment (TheMallen) [reply packages/react-form/src/hooks/field/test/field.test.tsx]: Because in one of the tests we use an object as the value and the HTML input haaaates it.  Maybe I'll just go back and make a different test component for that.

File: packages/predicates/CHANGELOG.md [hunk 6]
```diff
@@ -5,6 +5,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## [1.1.0]
+
+### Added
+
+- added the `notNumericString` predicate, which returns `true` when its input is not a numeric string and `false` otherwise.
```
Comment (cartogram) [inline packages/predicates/CHANGELOG.md]: Should we add a PR link?

File: packages/react-form/src/hooks/field/test/field.test.tsx [hunk 7]
```diff
@@ -194,6 +194,42 @@ describe('useField', () => {
...
+                onBlur={field.onBlur}
+              />
+            </label>
+            {field.error && <p>{field.error}</p>}
+          </>
+        );
+      }
+
+      const wrapper = mount(<FooBarTestField config={{foo: 'bar'}} />);
+      const newValue = faker.commerce.product();
```
Comment (cartogram) [inline packages/react-form/src/hooks/field/test/field.test.tsx]: that's handy
